Story/CCP-212

mkarthik90 opened the pull request:
Pagination and sorting changes for concept search screen. 
Will include test case once review comments are addressed.

mkarthik90 commented:
run integration tests please

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java:
```diff
@@ -206,6 +215,15 @@ public void insertConcept(ConceptEntry entry, String userName) throws LuceneExce
+                    if (searchFieldAnnotation.isSortAllowed()) {
+                        String value = StringEscapeUtils
+                                .unescapeHtml4(String.valueOf(contentOfField).replaceAll("\\s", "")).trim()
+                                .toLowerCase();
```
Couple of things here:
- it seems like this line is used more often; make that a method to call
- why do you remove whitespaces? 'hello world' is very different 'helloworld' and should be sorted to different positions
- html contents should be removed for sorting (the users only sees 'pony' and '**pony**' and would see why they would be on different positions.
- non-ascii characters like umlauts, accents, etc. should probably be turned into ascii for sorting (ö -> o, é -> e, etc.)

jdamerow commented on Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptSearchController.java:
```diff
@@ -53,28 +55,43 @@
             return "conceptsearch";
         }
+        
+        if (conceptIdsToMerge != null && !conceptIdsToMerge.equals("")) {
```
I don't think this check is really necessary. If the attribute is null, it's the same as when it's not set I believe.

mkarthik90 commented:
@jdamerow  Addressed review comments. For replacing the non-ascii characters with ascii characters, I replaced with empty string and sorting seems to work fine. Let me know if that's fine.

diging-jenkins commented:
Can one of the admins verify this patch?

diging-jenkins commented:
Can one of the admins verify this patch?

mkarthik90 commented:
@jdamerow  Addressed review comments.

jdamerow commented:
ok to test

jdamerow commented:
Couple of things:
- default sorting has to be by relevancy with exact matches first
- table headings need to have a little icon that shows the current sorting and direction (e.g. up/down arrow for ascending/descending, both for unsorted columns that can be sorted)

mkarthik90 commented:
@jdamerow Addressed review comments.

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented:
Ok, let's put some space between icon and column title so that it doesn't look so squished and we should be good to go.

mkarthik90 commented:
@jdamerow  Addressed review comment. Added spacing.
